hw/ip/spi_host/doc/_index.md

```diff
@@ -660,16 +659,613 @@ To use this reset, assert {{< regref "CONTROL.SW_RST" >}}, and then wait for the
+It also controls the timing of shift register operations, coordinating I/O on the `sd` bus with the other SPI signals.
+
+This section describes the SPI_HOST FSM and its control of the `sck` and `csb` lines as well as its interactions with the Shift Register and the Config/Command CDC.
+
+### Clock Divider
+
+The SPI_HOST FSM is driven by rising edge of the core clock, however the FSM state registers are not *enabled* during every cycle.
+There is an internal clock counter `clk_cntr_q` which repeatedly counts down from {{< regref "CONFIGOPTS.CLKDIV" >}} to 0, and the FSM is only enabled when `clk_cntr_q == 0`.
+
+The exception is when the FSM is one of the two possible Idle states (`Idle` or `IdleCSBActive`), in which case `clk_cntr_q` is constantly held at zero, making it possible to immediately transition out of the idle state as soon as a new command appears.
+Once the FSM transitions out of the idle state, `clk_cntr_q` resets to {{< regref "CONFIGOPTS.CLKDIV" >}}, and FSM transitions are only enabled at the divided clock rate.
+
+As shown in the waveform below, this has the effect of limiting the FSM transitions to only occur at discrete *timeslices* of duration:
+
+$$T_\textrm{timeslice} = \frac{T_{\textrm{clk},\textrm{core}}}{\texttt{clkdiv}+1}.$$
+
+{{< wavejson >}}
+{signal: [
+  {name: 'clk',        wave: 'p......................'},
+  {name: 'clkdiv',     wave: '2......................', data: "3"},
+  {name: 'clk_cntr_q', wave: '222222222222......22222', data: "3 2 1 0 3 2 1 0 3 2 1 0 3 2 1 0 3"},
+  {name: 'FSM state',  wave: '2...2.......2.....2...2', data: "WaitTrail WaitIdle Idle WaitLead Hi"              },
+  {name: 'fsm_en',     wave: '0..10......1......0..10'              },
+  {name: 'Timeslice Boundary', wave: "1...H...H...H.....H...H"}
+],
+  edge: ["A<->B min. 9 cycles", "C<->D min. 4 cycles"],
+ head: {text: "Use of FSM Enable Pulses to Realize Multi-Clock Timeslices", tock: 1},
+ foot: { text: "The fsm_en signal is always high in idle states, to allow exit transitions at any time"}
+}
+{{< /wavejson >}}
+
+#### Other Internal Counters
+
+In addition to the FSM state register, the SPI_HOST FSM block also has a number of internal registers to track the progress of a given command segment.
+
+- `wait_cntr_q`: This counter is used the hold the FSM in a particular state for several timeslices, in order to implement the `CSNIDLE`, `CSNLEAD` or `CSNTRAIL` delays required for a particular device.
+
+- `byte_cntr_q`, `bit_cntr_q`: These counters respectively track the number of bytes left to transmit for the current segment and the number of bits left to transmit in the current byte.
+
+- Finally, there are registers corresponding to each configuration field (`csid_q`, `cpol_q`, `cpha_`, `csnidle_q`, `csnlead_q`, `csntrail_q`, and `full_cyc_q`) and to each command segment field (`csaat`, `cmd_rd_en`, `cmd_wr_en`, and `cmd_speed`).
+This registers are sampled whenever a new command comes it, allowing the command inputs to change.
```

Comment:
Typo: "These registers" and "comes i**n**"

Reply:
```suggestion
This registers are sampled whenever a new command comes in, allowing the command inputs to change.
```